src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs

```diff
@@ -392,6 +392,13 @@ private DriveInfo ReportCommonDiskUsage(string dirType, string directory)
                     }
 #endif
                     Log.LogMessage("Success");
+
+                    // For additional directories we want only the contents deleted but not the root. The easiest is just to recreate the root folder
+                    // after the whole folder has been created. Since this just applies to additional ('known') folders we should not worry about long paths.
+                    if (isAdditionalDirectory && !Directory.Exists(directory))
```

Comment:
While this might help with some of the issues I don't think this is going to really help the real race condition. I suspect there are tmp files being written that need to remain as well.